pybamm/models/submodels/electrolyte/stefan_maxwell/diffusion/composite_stefan_maxwell_diffusion.py

```diff
@@ -49,24 +53,36 @@ def get_coupled_variables(self, variables):
         return variables
 
     def set_rhs(self, variables):
+        "Composite reaction-diffusion with source terms from leading order"
 
         param = self.param
 
         eps_0 = variables["Leading-order porosity"]
         deps_0_dt = variables["Leading-order porosity change"]
         c_e = variables["Electrolyte concentration"]
         N_e = variables["Electrolyte flux"]
-        source_terms_0 = sum(
-            pybamm.Concatenation(
-                reaction["Negative"]["s"]
-                * variables["Leading-order " + reaction["Negative"]["aj"].lower()],
-                pybamm.Broadcast(0, "separator"),
-                reaction["Positive"]["s"]
-                * variables["Leading-order " + reaction["Positive"]["aj"].lower()],
+        if self.extended is False:
```

Comment:
maybe separate functions might help readability here given that these are quite long expressions
```
if self.extended is False: 
    source_terms_0 = _get_unextended_source_terms()
elif: self.extended is True: 
    source_terms_0 = _get_extended_source_terms()
```